modules/reindex/src/test/java/org/elasticsearch/reindex/AsyncBulkByScrollActionTests.java

```diff
@@ -335,6 +342,48 @@ public void testBulkResponseSetsLotsOfStatus() throws Exception {
         }
     }
 
+    public void testHandlesBulkWithNoScroll() {
+        // given a request that should not open scroll
+        testRequest.setMaxDocs(1);
+        testRequest.getSearchRequest().source().size(100);
+
+        // when receiving bulk response
+        var responses = randomArray(0, 1, BulkItemResponse[]::new, AsyncBulkByScrollActionTests::createBulkResponse);
```

Comment:
I am not sure I understand why we test with array size 0 here?

Reply:
This verifies behavior if source index has no documents. 
This is relevant for a new condition added (exiting if source index has less documents then a max_docs)